packages/lit-dev-content/site/docs/ssr/authoring.md

```diff
@@ -0,0 +1,90 @@
+
+Note: The restrictions listed on this page are subject to change as we make improvements to Lit SSR. If you would like to see a certain use case supported, please [file an issue](https://github.com/lit/lit/issues/new/choose) or start a [discussion](https://github.com/lit/lit/discussions) thread.
+
+## Considerations
+
+### Browser only code
+
+Most browser DOM APIs are not available in the Node environment. Lit SSR utilizes a DOM shim that's the bare minimum required for rendering Lit templates and components. For a full list of what APIs are available, see the [DOM Emulation](/docs/ssr/dom-emulation) page.
+
+When authoring components, perform imperative DOM operations from lifecycle methods that are called only on the client, and not on the server. See the [lifecycle](#lifecycle) section below for what specific methods can be used.
+
+Some component modules may also have side effects that utilize browser APIs, for example to detect certain browser features, which would break simply by importing in a non-browser environment. If these side-effects cannot be moved to a client-only lifecycle, you must provide proper alternatives or some default behavior for when running in Node.
+
+For simple cases, adding conditionals or optional chaining to certain DOM accesses may be sufficient to guard against unavailable DOM APIs. For example:
+
+```js
+const hasConstructableStylesheets = typeof globalThis.CSSStyleSheet?.prototype.replaceSync === 'function';
+```
+
+The Lit team is also working on providing an easy way of checking whether the code is running in a Node environment which can be utilized to write conditional code blocks targeting different environments. Follow the issue [[labs/ssr] A way for client code to check if running in server (isSsr)](https://github.com/lit/lit/issues/3158) for updates on this feature.
+
+For more complex uses cases, consider utilizing [conditional exports](https://nodejs.org/api/packages.html#conditional-exports) in Node that specifically match for `"node"` environments so you could have different code depending on whether the module is being imported for use in Node or in the browser. Users would get the appropriate version of the package depending on whether it was imported from Node or the browser. Export conditions are also supported by popular bundling tools like [rollup](https://github.com/rollup/plugins/tree/master/packages/node-resolve#exportconditions) and [webpack](https://webpack.js.org/configuration/resolve/#resolveconditionnames) so users can bring in the appropriate code for your bundle.
+
+Note: Due to the way Lit libraries utilize conditional export for providing code meant for import in Node, we strongly discourage bundling `lit` into your component for packages being published to NPM as your bundle will only include a particular version of `lit` meant for a particular environment.
+
+### Lifecycle
+
```

Comment:
I wonder if we should add warning here that this set of server lifecycle callbacks is subject to change while SSR is in labs.

Reply:
I have that note specifically for `connectedCallback()` only, but that makes it seem like that's the only thing that could change so a generic warning might be better.

Reply:
I added the warning to the same aside mentioned https://github.com/lit/lit.dev/pull/880#discussion_r954342904 about methods called on server side should not contain DOM APIs. But maybe it's bad to say methods not called on server are safe, immediately followed by this is subject to change?

Reply:
I think these should be two separate messages. One is permanent, and not a warning: don't call DOM APIs in server lifecycle methods. The other is temporary and a warning: the set of client vs server lifecycle methods might chance while this library is unstable.

Reply:
Split it up and put the "subject to change" in a labs aside as that's meant to be the temporary one.